packages/rum-core/src/common/apm-server.js

```diff
@@ -124,53 +134,29 @@ class ApmServer {
   _makeHttpRequest(
     method,
     url,
-    { timeout = 10000, payload, headers, beforeSend } = {}
+    { timeout = HTTP_REQUEST_TIMEOUT, payload, headers, beforeSend } = {}
   ) {
-    return new Promise(function (resolve, reject) {
-      var xhr = new window.XMLHttpRequest()
-      xhr[XHR_IGNORE] = true
-      xhr.open(method, url, true)
-      xhr.timeout = timeout
-
-      if (headers) {
-        for (var header in headers) {
-          if (headers.hasOwnProperty(header)) {
-            xhr.setRequestHeader(header, headers[header])
-          }
+    // This bring the possibility of sending requests that outlive the page.
+    if (shouldUseFetchWithKeepAlive(method, payload)) {
+      return sendFetchRequest(method, url, {
+        keepalive: true,
+        timeout,
+        payload,
+        headers
```

Comment:
I'm checking how to do it, because in the fetch case we will not have the XHR object and probably we are facing a breaking change on this

Reply:
If we don't want to do a breaking change we should skip the keepalive logic if the beforeSend function is defined

Reply:
As agreed, if beforeSend is defined this new logic will not be executed otherwise we would creating a breaking change. In a different PR this will be handled